src/hooks/useOTPListener.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useRef } from 'react';
```

Comment:
i wonder if it would make sense to make PinInputField a folder like the TextField folder, then put the hook in there